src/field/FieldEquation.hh

```diff
@@ -0,0 +1,52 @@
+//---------------------------------*-CUDA-*----------------------------------//
+// Copyright 2020 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file FieldEquation.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "MagField.hh"
+#include "base/Types.hh"
+#include "physics/base/Units.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * The FieldEquation evaluates the righ hand side of the Lorentz equation
+ * for a given magnetic field value.
+ */
+class FieldEquation
+{
+  public:
+    //@{
+    //! Type aliases
+    using OdeState = Array<real_type, 6>;
+    //@}
+
+  public:
+    //! Construct with a magnetic field
+    CELER_FUNCTION FieldEquation(const MagField& field);
+
+    //! Set charge if q != -eplus
+    CELER_FUNCTION void set_charge(units::ElementaryCharge q);
```

Comment:
As I mentioned on the previous PR, this should not be a setter function. Your `FieldEquation` is essentially acting like a `TrackView` based on the track "state" (the charge) and the field "params" (the parameterization -- in this case, hard-coded as uniform). Change the setter to be part of the constructor since this class is only going to be instantiated as a temporary helper class. I think it would also help to change the class name to indicate that it's a convolution of the particle state (charge) and the magnetic field equation definition.

<!--
-- it doesn't make sense in a GPU context where we're instantiating a field equation class for a single track (indeed a single step within that track). I suggest one of the three approaches for `FieldEquation`:
1. Change to  becomes a `TrackView`-style class that supports multiple field representations (e.g. constant, r-z binned, constant-in-volume, or parameterized) that can be different for every track. The charge becomes part of the constructor, as well as the field type and other necessary properties such as volume ID.
2. Change to a `TrackView` style class that could be templated on a `FieldType` which is the parameterization (constant, binned, etc.). Particle charge is passed to the constructor along with shared Params data such as the spatial parameterization.
3. Instead of the charge being part of the FieldEquation (charge isn't part of the "equation" anyway) it becomes part of the FieldState, so your `OdeArray` gains an extra scalar which could be excluded from the `axpy`.
-->

Reply:
Move the charge into the constructor and eemoved the setter. Rename the class name to MagFieldEquation.